src/metabase/api/task.clj

```diff
@@ -0,0 +1,26 @@
+(ns metabase.api.task
+  "/api/task endpoints"
+  (:require
+    [compojure.core :refer [GET]]
+    [metabase
+     [util :as u]]
+    [metabase.api
+     [common :as api]]
+    [metabase.models
+     [task-history :as tasks :refer [TaskHistory]]]
+    [toucan
+     [db :as db]]))
+
+
+(api/defendpoint GET "/"
+  "Fetch a list of recent tasks stored as Task History"
+  []
+  (as-> (db/select TaskHistory) tasks))
+
+(api/defendpoint GET "/:id"
+  "Get `TaskHistory` entry with ID."
+  [id]
+  (u/prog1 (-> (TaskHistory id)
+               api/read-check)))
```

Comment:
Looks like I can't "suggest" with two lines. What you have will work, but this is an easier way to say it:
```
(api/read-check TaskHistory id)
```
That will check for existence, permissions and query the DB.